hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/HoodieSparkEngineContext.java

```diff
@@ -260,7 +260,7 @@ public <I, O> O aggregate(HoodieData<I> data, O zeroValue, Functions.Function2<O
   public ReaderContextFactory<?> getReaderContextFactory(HoodieTableMetaClient metaClient) {
     // metadata table are only supported by the AvroReaderContext.
     if (metaClient.isMetadataTable()) {
-      return new AvroReaderContextFactory(metaClient);
+      return new AvroReaderContextFactory(metaClient, new TypedProperties());
```

Comment:
So this, used by the metadata writer, cannot leverage the cache configs?

Reply:
Metadata writer can use leverage cache configs. It is not currently using this method for creating an `AvroReaderContextFactory`.

Instead, it is using:
```
org.apache.hudi.common.engine.HoodieEngineContext#getReaderContextFactoryForWrite
```

Which is able to pass in hfile cache configs. The other callers that are using `AvroReaderContextFactory((metaClient, new TypedProperties())` will have their behaviour remain unchanged.